gamemodes/terrortown/gamemode/server/sv_main.lua

```diff
@@ -953,7 +958,19 @@ function GM:TTT2PrePrepareRound(duration)
     if cvPreferMapModels:GetBool() and self.force_plymodel and self.force_plymodel ~= "" then
         self.playermodel = self.force_plymodel
     elseif cvSelectModelPerRound:GetBool() then
-        self.playermodel = playermodels.GetRandomPlayerModel()
+        if cvSelectUniqueModelPerPlayer:GetBool() then
+            local plys = player.GetAll()
+            for i = 1, #plys do
+                plys[i].defaultModel = playermodels.GetRandomPlayerModel()
+            end
+        else
+            local plys = player.GetAll()
+            for i = 1, #plys do
+                plys[i].defaultModel = nil
+            end
+
+            self.playermodel = playermodels.GetRandomPlayerModel()
+        end
```

Comment:
@TimGoll 
Can we find out what addons etc. write to `.defaultModel`? Because until now we (ttt2) never wrote to it and only read it.

If we start writing to and even resetting(!) it that might be a problem. Even more so now that the default behaviour is changed here.

Don't get me wrong this could all be completely fine, but i have no idea where any value for `.defaultModel` would have come from. Certainly not from ttt2

Reply:
I had the same thoughts, however I'm unable to find any use of that variable in our whole code base. My test server also has loads of addons and I didn't encounter any issues. My suspicion is that this is a very old part of the code that is no longer used. But I could also introduce a new variable for that.

Reply:
`.defaultModel` is used twice in our codebase.

[First occurrence](https://github.com/TTT-2/TTT2/blob/b03a912682a572b6b3b9e341f8662f2c62e020e7/gamemodes/terrortown/gamemode/shared/sh_player_ext.lua#L1105) in `plymeta:SetModel(mdlName)` as a fallback if the given `mdlName` is not a valid model.

[Second occurrence](https://github.com/TTT-2/TTT2/blob/b03a912682a572b6b3b9e341f8662f2c62e020e7/gamemodes/terrortown/gamemode/server/sv_player.lua#L268) which calls the function in the first occurrence with `ply.defaultModel or GAMEMODE.playermodel`. Which is somewhat stupid in itself ...

If we can reasonably believe that this does not break playermodelselector addons i'm fine with merging this. streamlining the playermodel handling can come later.

Reply:
I meant that it is set nowhere. But you are right. I might ask some EPMS users on discord to test this branch

Reply:
![image](https://github.com/TTT-2/TTT2/assets/13639408/b2a1d064-f358-467d-8e15-10d65c68918b)

seems to work

Reply:
@Histalek I think you probably missed this comment